statesync: test flake on TestSyncer_SyncAny

## Repro:

```shell
go test -count=10 -race -run TestSyncer_SyncAny'$' ./internal/statesync
```

I think `-race` is not required to reproduce this.

## Observe:

```
--- FAIL: TestSyncer_SyncAny (122.01s)
    syncer_test.go:166: added chunk (1 of 4): 0
    syncer_test.go:166: added chunk (2 of 4): 1
    syncer_test.go:166: added chunk (3 of 4): 2
    syncer_test.go:198: 
        	Error Trace:	syncer_test.go:198
        	Error:      	Received unexpected error:
        	            	no suitable snapshots found
        	Test:       	TestSyncer_SyncAny
    syncer_test.go:148: sent 3 chunks
FAIL
FAIL	github.com/tendermint/tendermint/internal/statesync	184.145s
```

--- a/internal/statesync/syncer_test.go
+++ b/internal/statesync/syncer_test.go
@@ -173,7 +173,7 @@ func TestSyncer_SyncAny(t *testing.T) {
 	// beginning. We also wait for a little while, to exercise the retry logic in fetchChunks().
 	connSnapshot.On("ApplySnapshotChunk", mock.Anything, abci.RequestApplySnapshotChunk{
 		Index: 2, Chunk: []byte{1, 1, 2},
-	}).Once().Run(func(args mock.Arguments) { time.Sleep(2 * time.Second) }).Return(
+	}).Once().Run(func(args mock.Arguments) { time.Sleep(1 * time.Second) }).Return(
 		&abci.ResponseApplySnapshotChunk{
 			Result:        abci.ResponseApplySnapshotChunk_RETRY_SNAPSHOT,
 			RefetchChunks: []uint32{1},

